WURFL/Storage/Redis.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * Copyright (c) 2015 ScientiaMobile, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Refer to the COPYING.txt file distributed with this package.
+ *
+ * @category   WURFL
+ * @package	WURFL_Storage
+ * @copyright  ScientiaMobile, Inc.
+ * @license	GNU Affero General Public License
+ * @author	 Fantayeneh Asres Gizaw
+ * @version	$id$
+ */
+/**
+ * WURFL Storage
+ * @package	WURFL_Storage
+ */
+class WURFL_Storage_Redis extends WURFL_Storage_Base {
+
+    const EXTENSION_MODULE_NAME = "redis";
+
+    private $defaultParams = array(
+        "host" => "127.0.0.1",
+        "port" => "6379",
+        "hash_name" => 'WURFL_DATA',
+        "redis" => null,
+        "database" => 0
+    );
+
+    private $database;
+    private $host;
+    private $port;
+    private $redis;
+    private $hashName;
+
+    protected $supports_secondary_caching = true;
+
+    public function __construct($params) {
+        $currentParams = is_array($params)? array_merge($this->defaultParams, $params): $this->defaultParams;
+        $this->initialize($currentParams);
+    }
+
+    public function initialize($params) {
+        $this->_ensureModuleExistence();
+        $this->host = $params['host'];
+        $this->port = $params['port'];
+        $this->hashName = $params['hash_name'];
+        $this->database = $params['database'];
+        if (!is_null($params['redis']) && ($params['redis'] instanceof Redis)) {
+            $this->redis = $params['redis'];
```

Comment:
Yes I thought of this as an option for the user to just inject a fully prepared Redis object (for example, when Redis is in a service container). This object should be already connected and with the correct database selected.